src/cds/src/main/kotlin/org/icpclive/cds/cats/CATSDataSource.kt

```diff
@@ -0,0 +1,175 @@
+package org.icpclive.cds.cats
+
+import kotlinx.coroutines.*
+import kotlinx.datetime.Clock
+import kotlinx.serialization.Serializable
+import org.icpclive.api.*
+import org.icpclive.cds.ContestParseResult
+import org.icpclive.cds.FullReloadContestDataSource
+import org.icpclive.cds.common.jsonLoaderService
+import org.icpclive.util.getCredentials
+import java.awt.Color
+import java.util.*
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.seconds
+
+
+class CATSDataSource(val properties: Properties, creds: Map<String, String>) : FullReloadContestDataSource(5.seconds) {
+    private val login = creds["cats.login"]
+    private val password = creds["cats.password"]
+
+    // :TODO I may send less count of quires to the server but I need to know when my sid is expired
+    private val sid: String
+        get() = runBlocking {
+            async { authLoader.loadOnce().sid }.await()
+        }
+
+    @Serializable
+    data class Auth(val status: String, val sid: String, val cid: Long)
+
+    @Serializable
+    data class Problem(val id: Int, val name: String, val code: String)
+
+    @Serializable
+    data class Problems(val problems: ArrayList<Problem>)
+
+    @Serializable
+    data class Team(val id: Int, val account_id: Int, val login: String, val name: String, val role: String)
+
+    @Serializable
+    data class Users(val users: ArrayList<Team>)
+
+    @Serializable
+    data class Contest(
+        val title: String,
+        val start_date: String,
+        val freeze_date: String,
+        val finish_date: String,
+        val rules: String
+    )
+
+    @Serializable
+    data class Run(
+        val type: String,
+        val id: Int? = null,
```

Comment:
Does it really possible not to have this field?

Reply:
If it's talking about 'type' then no, it's necessary field

Reply:
![image](https://user-images.githubusercontent.com/54590439/211147343-5e3634a5-b235-48df-bbff-9ec2bf096895.png)

So we have in one array, these types of objects
submit and broadcast. We need to read only submit

Reply:
It can be implemented as 

```
@Serializable
sealed class Run
@Serializable
@SerialName("submit")
class Submit : Run { ... }
@Serializable
@SerialName("broadcast")
class Broadcast : Run { ... }
```

But this is really strange API